import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+        "meta_resource_sample.txt",
+        "meta_resource_study.txt",
+        "meta_rppa.txt",
+        "meta_rppa_Zscores.txt",
+        "meta_timeline.txt"
+        ]
+    filenames_in_use_before_update_set = set()
+    wildcard_patterns_in_use_before_update_set = set()
+    for filename in FILENAMES_IN_USE_BEFORE_UPDATE_LIST:
+        if "*" in filename:
+            wildcard_patterns_in_use_before_update_set.add(filename.replace(CANCER_STUDY_TAG(), cancer_study_identifier))
+        else:
+            filenames_in_use_before_update_set.add(filename.replace(CANCER_STUDY_TAG(), cancer_study_identifier))
+    return filenames_in_use_before_update_set, wildcard_patterns_in_use_before_update_set
+
+def parse_current_datatypes(cancer_study_identifier):
+    """
+    the contents of the datatype sheet (in tab separated format) are parsed and all filenames (both meta and staging) or wildcard patterns are mapped to their associated datatypes. The maps are returned.
+    """
+    datatype_sheet_lines = datatype_sheet_download_tab_separated().splitlines()
+    filename2datatype = {}
+    wildcard_pattern2datatype = {}
+    header_skipped = False
+    for line in datatype_sheet_lines:
+        if not header_skipped:
+            if line.startswith("datatype"):
+                header_skipped = True
+            continue
+        fields = line.split("\t")
+        datatype = fields[0].strip()
+        staging_filename = fields[5].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        meta_filename = fields[9].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        if "*" in staging_filename:
+            wildcard_pattern2datatype[staging_filename] = datatype
+        else:
+            filename2datatype[staging_filename] = datatype
+        if "*" in meta_filename:
+            wildcard_pattern2datatype[meta_filename] = datatype
+        else:
+            filename2datatype[meta_filename] = datatype
+    return filename2datatype, wildcard_pattern2datatype
+
+def is_normal_sample_filename(filename):
+    """
+    examines filename to see if the importer would skip it as a "normal sample" file.
+    """
+    normal_sample_suffixes = ["_normals.txt", "data_mrna_seq_v2_rsem_normal_samples.txt", "data_mrna_seq_v2_rsem_normal_samples_zscores_ref_normal_samples.txt"]
+    for suffix in normal_sample_suffixes:
+        if (filename.endswith(suffix)):
+            return True;
+    return False;
+
+def determine_cancer_study_identifier_from_meta_file(filepath):
+    """
+    search file at <filepath> for a cancer_study_identifier and return it if found
+    """
+    CANCER_STUDY_IDENTIFIER_PREFIX = "cancer_study_identifier:"
+    file = open(filepath, 'r')
+    for line in file.readlines():
+        line_without_whitespace = "".join(line.split()).strip()
+        if line.startswith(CANCER_STUDY_IDENTIFIER_PREFIX):
+            cancer_study_identifier = line[len(CANCER_STUDY_IDENTIFIER_PREFIX):].strip()
```

Comment:
maybe we shoud just split the line on colons? and take the second part? e.g line.split(':')[1]

Reply:
something like 
```
cancer_study_identifier = line.split(":")[1].strip() (there should be a strip for both front and back)
if cancer_study_identifier:
blahblahblahblabha
```

Reply:
Thinking about the possibility of multiple colons on the line (some colons occurring in the value), we would have to join together all parts of the list after the first element, such as
":".join(split_line_fields[1:])

So we chose to recode this by splitting at the position of the first occurrence of ":"

Reply:
After looking again, the code already discards all whitespace on the line (from the meta file). It does this before tests are performed, so the prefix "cancer_study_identifier:" would be seen and could be stripped off of the line based on length, which would yield just the value string (minus any spaces). Since cancer studies should have no spaces, there should be no problem. So I think I'll keep the code as is.